src/model/remoteConfig.ts

```diff
@@ -168,134 +190,181 @@ export const DEFAULT_CONFIG: RainbowConfig = {
+}>;
 
-const QUERY_PARAMS = {
-  queryKey: remoteConfigQueryKey,
-  queryFn: fetchRemoteConfig,
-  staleTime: 600_000, // 10 minutes
-  placeholderData: DEFAULT_CONFIG,
-  retry: 3,
-  retryDelay: (attempt: number) => Math.min(attempt > 1 ? 2 ** attempt * 1000 : 1000, 30 * 1000),
+const STRINGIFIED_FIREBASE_DEFAULTS: StringifiedFirebaseDefaults = {
+  default_slippage_bips: JSON.stringify(DEFAULT_CONFIG.default_slippage_bips),
+  default_slippage_bips_chainId: JSON.stringify(DEFAULT_CONFIG.default_slippage_bips_chainId),
 };
 
+type FirebaseConfigDefaults = Omit<RainbowConfig, keyof StringifiedFirebaseDefaults> &
+  StringifiedFirebaseDefaults &
+  FirebaseRemoteConfigTypes.ConfigDefaults;
+
+/**
+ * Builds the payload for `rc.setDefaults` with stringified objects.
+ */
+function getFirebaseDefaults(): FirebaseConfigDefaults {
+  return { ...DEFAULT_CONFIG, ...STRINGIFIED_FIREBASE_DEFAULTS };
+}
+
+// ============ Remote Config Store ============================================ //
+
+interface RemoteConfigState {
+  config: RainbowConfig;
+  lastFetchedVersion: number;
+  getRemoteConfigKey: <K extends keyof RainbowConfig>(key: K) => RainbowConfig[K];
+}
+
+export const useRemoteConfigStore = createQueryStore<RainbowConfig, never, RemoteConfigState>(
+  {
+    fetcher: fetchRemoteConfig,
+    retryDelay: (retryCount: number) => Math.min(retryCount > 1 ? 2 ** retryCount * 1_000 : 1_000, 30_000),
+    setData: ({ data, set }) =>
+      set(state => {
+        const configChanged = !dequal(state.config, data);
+        const versionChanged = state.lastFetchedVersion !== REMOTE_CONFIG_VERSION;
+
+        return configChanged || versionChanged
+          ? {
+              config: configChanged ? data : state.config,
+              lastFetchedVersion: REMOTE_CONFIG_VERSION,
+            }
+          : state;
+      }),
+    cacheTime: time.weeks(1),
+    maxRetries: 3,
+    staleTime: time.minutes(10),
+  },
+
+  (_, get) => ({
+    config: DEFAULT_CONFIG,
+    lastFetchedVersion: 0,
+    getRemoteConfigKey: key => get().config[key],
+  }),
+
+  { storageKey: 'remoteConfig' }
+);
+
+// ============ Public Methods ================================================= //
+
 export async function initializeRemoteConfig(): Promise<void> {
-  const rc = remoteConfig();
-  await rc.setConfigSettings({
-    minimumFetchIntervalMillis: 120_000,
-  });
-  const defaults: Record<string, string | number | boolean> = {
-    ...DEFAULT_CONFIG,
-    default_slippage_bips: JSON.stringify(DEFAULT_CONFIG.default_slippage_bips),
-    default_slippage_bips_chainId: JSON.stringify(DEFAULT_CONFIG.default_slippage_bips_chainId),
-  };
-  await rc.setDefaults(defaults);
-  await Promise.race([queryClient.prefetchQuery(QUERY_PARAMS), delay(3000)]);
+  const { fetch, lastFetchedVersion } = useRemoteConfigStore.getState();
+  if (lastFetchedVersion !== REMOTE_CONFIG_VERSION) {
+    await Promise.race([fetch(), delay(time.seconds(3))]);
+  }
 }
```

Comment:
Feel like force close to get latest config is a common thing, and this prevents that from happening. You will have to wait some unknown stale time duration before you can test a config change went through.

Reply:
Now fetching every cold start but only waiting on it when the version changes